libs/LzmaLib/source/7zTypes.h

```diff
@@ -6,6 +6,20 @@
 
 #ifdef _WIN32
 /* #include <windows.h> */
+#else
+	// Undefined types/definitions that GCC needs to compile on modern 
+	// MacOS/linux, there is probably a better way of detecting this.
+	#ifndef LPVOID 
+		#define LPVOID void*
+		#define HRESULT long
+		#define BOOL _Bool
+		#define LONG unsigned long
+
+		#define S_OK 0
+		#define E_FAIL 1
+		#define TRUE True
+		#define FALSE False
+	#endif
```

Comment:
```suggestion
	#include <stdbool.h>

	typedef int32_t		LONG;
	typedef uint32_t	ULONG;
	typedef void*		LPVOID;
	typedef int32_t		HRESULT;

	#ifndef S_OK
		#define S_OK	((HRESULT)0x00000000L)
	#endif

	#ifndef E_FAIL
		#define E_FAIL	((HRESULT)0x80000008L)
	#endif
```

I couldn't quickly find references to `LONG` in the code just using review tool.
It would be great to remove it and let the preprocessor directives below handle it, if possible :)

Reply:
Much nicer! Needs `#include <stdint.h>`, but otherwise perfect.

Reply:
I did end up defining BOOL,TRUE,FALSE again, gcc `stdbool` only sets lowercase. `LONG` has been removed from the offending line, replaced with `Int32`.